test/integration/health_check_integration_test.cc

```diff
@@ -673,15 +671,14 @@ TEST_P(TcpHealthCheckIntegrationTest, DisableHCForActiveTraffic) {
   EXPECT_EQ(1, test_server_->counter("cluster.cluster_1.health_check.success")->value());
   EXPECT_EQ(0, test_server_->counter("cluster.cluster_1.health_check.failure")->value());
 
-  timeSystem().advanceTimeWait(std::chrono::seconds(1));
   test_server_->waitForCounterEq("cluster.cluster_1.health_check.attempt", 3);
 
   ASSERT_TRUE(cluster_data.host_fake_raw_connection_->waitForData(
       FakeRawConnection::waitForInexactMatch("Ping")));
   result = clusters_[cluster_idx].host_fake_raw_connection_->write("Pong");
   RELEASE_ASSERT(result, result.message());
 
-  timeSystem().advanceTimeWait(std::chrono::milliseconds(10));
+  test_server_->waitForCounterEq("cluster.cluster_1.health_check.success", 2);
```

Comment:
You can remove the next line as it is redundant. Please also run all the new tests and check for flakes per the instructions here: https://github.com/envoyproxy/envoy/tree/main/test/integration#reproducing-test-flakes

/wait